css/amp.css

```diff
@@ -153,6 +153,14 @@ html.i-amphtml-ios-embed.i-amphtml-ios-overscroll > #i-amphtml-wrapper {
   position: relative;
 }
 
+/**
+ * Children of responsive elements should not be able to influence the height
+ * which they do when they are statically positioned.
+ */
+.i-amphtml-layout-responsive > :not(i-amphtml-sizer) {
```

Comment:
`i-amphtml-layout-size-defined` already has `overflow: hidden` which should enforce this. `i-amphtml-layout-responsive` and `i-amphtml-layout-size-defined` are added at the same time by core JS. 

If https://github.com/ampproject/amphtml/issues/9397#issuecomment-302243409 and https://github.com/ampproject/amphtml/issues/9397#issuecomment-302244538 indicate the root cause of #9397, I think this needs to be
```
[layout=responsive] {
 overflow: hidden !important;
}
```

instead.